packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieveMetadata/describers.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { LocalComponent } from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import { join } from 'path';
+import { RetrieveDescriber } from '.';
+import { BrowserNode } from '../../orgBrowser';
+import { SfdxPackageDirectories } from '../../sfdxProject';
+
+abstract class NodeDescriber implements RetrieveDescriber {
+  protected node: BrowserNode;
+
+  constructor(node: BrowserNode) {
+    this.node = node;
+  }
+
+  public abstract buildMetadataArg(): string;
+
+  public abstract gatherOutputLocations(): Promise<LocalComponent[]>;
+
+  protected async buildOutput(node: BrowserNode): Promise<LocalComponent[]> {
+    const typeNode = node.getAssociatedTypeNode();
+    // create cmp for every package directory to accommodate for the cli retrieve bug
+    const packageDirectories = await SfdxPackageDirectories.getPackageDirectoryPaths();
+    return packageDirectories.map(directory => ({
```

Comment:
Because of the CLI bug where sometimes a component can be retrieved to a non-default package directory, we need to check every one of them and add it as a separate component. Once the bug is fixed we can remove this

Reply:
We should probably add the work item number to the comment on the code